contrib/opencensus-ext-azure/opencensus/ext/azure/integration/azure_functions.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from logging import Logger
+from typing import List, Dict, Optional
+
+from azure.functions import (
+    Context,
+    AppExtensionBase,
+    FunctionExtensionException
+)
+
+from opencensus.trace import config_integration
+from opencensus.trace.propagation.trace_context_http_header_format import (
+    TraceContextPropagator
+)
+from opencensus.trace.samplers import ProbabilitySampler
+from opencensus.trace.tracer import Tracer
+from ..trace_exporter import AzureExporter
+
+
+# The app setting name for providing default opencensusu appinsights key
+DEFAULT_INSTRUMENTATION_KEY_SETTING = 'APPINSIGHTS_INSTRUMENTATIONKEY'
+
+
+class OpenCensusExtension(AppExtensionBase):
+    @classmethod
+    def init(cls):
+        cls._tracers: Dict[str, Tracer] = {}
+        cls._exporter: Optional[AzureExporter] = None
+        cls._trace_integrations: List[str] = []
+        cls._default_key: Optional[str] = os.environ.get(
+            DEFAULT_INSTRUMENTATION_KEY_SETTING
+        )
+
+    @classmethod
+    def configure(cls,
+                  libraries: List[str],
+                  instrumentation_key: Optional[str] = None,
+                  *args,
+                  **kwargs):
+        """Configure the OpenCensus extension
+        :type libraries: List[str]
+        :param libraries: the libraries that need to be integrated into
+            OpenCensus tracer. (e.g. ['requests'])
+        :type instrumentation_key: Optional[str]
+        :param instrumentation_key: the instrumentation key for azure exporter
+            to write into. If this is set to None, the extension will write to
+            the AppInsight resource defined in APPINSIGHTS_INSTRUMENTATIONKEY
+        """
+        cls._trace_integrations = config_integration.trace_integrations(
+            libraries
+        )
+
+        if not instrumentation_key and not cls._default_key:
+            raise FunctionExtensionException(
+                'Please ensure either instrumentation_key is passed into '
+                'OpenCensusExtension.configure() method, or the app setting '
+                'APPINSIGHTS_INSTRUMENTATIONKEY is provided.'
+            )
+
+        cls._exporter = AzureExporter(
+            instrumentation_key=instrumentation_key or cls._default_key
+        )
+
+    @classmethod
+    def pre_invocation_app_level(cls,
+                                 logger: Logger,
```

Comment:
Please add a docstring or link to the docs in `azure.functions` library to provide more details on what these parameters are.

Reply:
This should be already added in azure-functions SDK. https://github.com/Azure/azure-functions-python-library/blob/dev/azure/functions/extension/app_extension_base.py#L65-L88

Reply:
Will add a link to app_extension_base.py